### PR TITLE
391 ispyb interaction and comment

### LINKED ISSUE #390: Smargon: Adding ioc name twice
During testing we found that the smargon would add the IOC name twice to the PV, this would mean that it failed to connect to the PVs.

## Acceptance Criteria
* The smargon is correctly created so that PVs names are correct
* We have a test in place to confirm this, probably a system test

### LINKED ISSUE #391: Adding error to Ispyb comment fails
During testing we got the error:
```
  File "/dls_sw/i03/software/artemis/src/artemis/external_interaction/ispyb/store_in_ispyb.py", line 59, in end_deposition
    self.update_grid_scan_with_end_time_and_status(
  File "/dls_sw/i03/software/artemis/src/artemis/external_interaction/ispyb/store_in_ispyb.py", line 104, in update_grid_scan_with_end_time_and_status
    params["comments"] += f" {run_status} reason: {reason}"
TypeError: unsupported operand type(s) for +=: 'NoneType' and 'str'
```
After we had an error in creating the zocalo connection

## Acceptance Criteria
* The above exception is fixed
* There is a unit test for this

### PR DESCRIPTION
**Pull request by dperl-dls**
Fixes #391 #393 
Incidentally fixes #390 
Includes changes from #400
Loads existing comment from ispyb and appends if there was an error

### To test:
1. Run tests
2. Ideally, run a grid scan with S03, and the dev ispyb config in the $ISPYB_CONFIG_PATH env variable, inspect the logs and see that the comment is loaded

### CODE REVIEW

**Review by DominicOram — CHANGES_REQUESTED** [review]
Looks good, nice and neat
* I'm not sure if we should keep `TESTING_smargon` in? If we do can you add a comment on https://github.com/DiamondLightSource/python-artemis/issues/377 to remove it?
* The sqlalchemy interface to ispyb is nice, can you make an issue to look into using it for writing?
* I added a system test for the ispyb interaction, can you review it for me

File: src/artemis/external_interaction/communicator_callbacks.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ class ISPyBHandlerCallback(CallbackBase):
     def __init__(self, parameters: FullParameters):
         self.params = parameters
         self.descriptors: Dict[str, dict] = {}
-        ispyb_config = os.environ.get("ISPYB_CONFIG_PATH", "TEST_CONFIG")
+        ispyb_config = os.environ.get("ISPYB_CONFIG_PATH", SIM_ISPYB_CONFIG)
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/artemis/external_interaction/communicator_callbacks.py:66]: done

File: src/artemis/external_interaction/ispyb/store_in_ispyb.py [hunk 2]
```diff
@@ -84,6 +96,23 @@ def store_grid_scan(self, full_params: FullParameters):
     def _store_scan_data(self):
         pass
 
+    def get_current_datacollection_comment(self, dcid: int) -> str:
+        """Read the 'comments' field from the given datacollection id ISPyB entry.
+        Returns an empty string if the comment is not yet initialised.
+        """
+        LOGGER.debug("Getting comment from ISPyB")
+        with ispyb.open(self.ISPYB_CONFIG_FILE) as self.conn:
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/artemis/external_interaction/ispyb/store_in_ispyb.py:104]: nope, dunno how that snuck in there

File: src/artemis/external_interaction/ispyb/store_in_ispyb.py [hunk 3]
```diff
@@ -84,6 +96,23 @@ def store_grid_scan(self, full_params: FullParameters):
     def _store_scan_data(self):
         pass
 
+    def get_current_datacollection_comment(self, dcid: int) -> str:
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/artemis/external_interaction/ispyb/store_in_ispyb.py:99]: yep

**Comment by dperl-dls** [comment]
I would like to keep the TESTING_I03Smargon device for now, I tend to do a lot of testing with S03 and it might take a while before we have added the PVs. Made a note to remove it.

File: src/artemis/external_interaction/ispyb/store_in_ispyb.py [hunk 4]
```diff
@@ -84,6 +96,23 @@ def store_grid_scan(self, full_params: FullParameters):
...
+        LOGGER.debug("Getting comment from ISPyB")
+        with ispyb.open(self.ISPYB_CONFIG_FILE) as self.conn:
+            with TRACER.start_span("read_comment_from_ispyb"):
+                with self.Session() as session:
+                    query = session.query(DataCollection).filter(
+                        DataCollection.dataCollectionId == dcid
+                    )
+                    current_comment: str = query.first().comments
+        if current_comment is None:
+            current_comment = ""
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/artemis/external_interaction/ispyb/store_in_ispyb.py:117]: - Added case of nothing coming back to the system test
- Logging and swallowing exceptions in `get_current_datacollection_comment`
- Added a unit test for this

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**